framework/src/media/MediaRecorderImpl.cpp

```diff
@@ -273,29 +278,29 @@ recorder_result_t MediaRecorderImpl::stop()
 	medvdbg("MediaRecorderImpl::stop()\n");
 	RecorderWorker& mrw = RecorderWorker::getWorker();
 	if (!mrw.isAlive()) {
-		return RECORDER_ERROR;
+		return RECORDER_NOT_ALIVE;
 	}
-	mrw.enQueue(&MediaRecorderImpl::stopRecorder, shared_from_this(), 0);
 
+	mrw.enQueue(&MediaRecorderImpl::stopRecorder, shared_from_this(), RECORDER_OK);
 	return RECORDER_OK;
 }
 
-void MediaRecorderImpl::stopRecorder(int errcode)
+void MediaRecorderImpl::stopRecorder(recorder_result_t ret)
 {
 	medvdbg("stopRecorder mCurState : %d\n", (recorder_state_t)mCurState);
 
-	if (errcode == 0 && mCurState != RECORDER_STATE_RECORDING && mCurState != RECORDER_STATE_PAUSED) {
-		notifyObserver(OBSERVER_COMMAND_ERROR);
-		meddbg("stopRecorder Failed mCurState : %d\n", (recorder_state_t)mCurState);
-		return;
-	}
+	if (ret == RECORDER_OK)
+	{
+		if (mCurState != RECORDER_STATE_RECORDING && mCurState != RECORDER_STATE_PAUSED) {
+			notifyObserver(OBSERVER_COMMAND_STOP_ERROR, RECORDER_INVALID_STATE);
+			meddbg("stopRecorder Failed mCurState : %d\n", (recorder_state_t)mCurState);
+			return;
+		}
 
-	audio_manager_result_t result = stop_audio_stream_in();
-	if (result != AUDIO_MANAGER_SUCCESS) {
-		meddbg("stop_audio_stream_in failed ret : %d\n", result);
-		if (errcode == 0) {
-			/* TODO : inform device control fail */
-			notifyObserver(OBSERVER_COMMAND_ERROR);
+		audio_manager_result_t result = stop_audio_stream_in();
```

Comment:
This should be called if it is not RECORDER_OK.

Reply:
no. It can be called in two cases.